### PR TITLE
#562 - Clear symptoms on uninstall, reset language on leave

### PR DESCRIPTION
**Pull request by AlanSl**
For #562

Previously `SecureStore` had been chosen for symptoms checker history. This isn't ideal because users uninstalling the app would be unlikely to expect their symptom history to persist (although users selecting "leave" in settings will still have their symptoms history removed). SecureStore is also not generally intended for growing data like daily symptoms history.

This migrates symptoms history to AsyncStorage, checking for legacy data in SecureStore and merging and de-duping it if there is any. It seems robust from my testing, including edge cases like testflight users downgrading their app to versions before this PR, but **please do test thoroughly**!

This PR also tidies up a few things related to storing state:

 - It separates `StorageKeys` by type and automatically clears all keys rather than relying on each being manually added to `clearContext`. This had been a common source of bugs.
 - It resets app language to device language on clearing app data

### CODE REVIEW

File: src/providers/context.tsx [hunk 1]
```diff
@@ -184,10 +187,36 @@ export const AP = ({appConfig, user, consent, children}: API) => {
...
+        // Before v1.1.0 release, symptoms data was stored in SecureStore
+        const [checksData, legacyChecksData] = await Promise.all([
+          AsyncStorage.getItem(AsyncStorageKeys.symptomKeys),
+          SecureStore.getItemAsync(AsyncStorageKeys.symptomKeys)
+        ]);
+        if (legacyChecksData) {
+          await SecureStore.deleteItemAsync(AsyncStorageKeys.symptomKeys);
+
+          // Merge and de-dupe to ensure no history is lost in version downgrades etc
+          checks = [
```
Comment (floridemai) [inline src/providers/context.tsx:199]: When will we have data in both of them?

File: src/providers/context.tsx [hunk 2]
```diff
@@ -184,10 +187,36 @@ export const AP = ({appConfig, user, consent, children}: API) => {
...
+
+          await AsyncStorage.setItem(
+            AsyncStorageKeys.symptomKeys,
+            JSON.stringify(checks)
+          );
+        } else {
+          checks = checksData ? JSON.parse(checksData) : [];
+        }
+
+        console.log({checks, checksData, legacyChecksData});
```
Comment (floridemai) [inline src/providers/context.tsx]: Remove console?

**Review by floridemai — COMMENTED** [review]
LGTM

File: src/providers/context.tsx (hunk 1 above)
Comment (AlanSl) [reply src/providers/context.tsx:199]: Just to cover edge cases, e.g. a testflight user or stakeholder switches between before and after versions. If we cover any such case we don't need to figure out every scenario and whether they're likely or not because they're all covered.

**Review by floridemai — APPROVED** [review]
